syscalls_darwin.go

```diff
@@ -60,8 +61,13 @@ type sockaddrCtl struct {
 var sockaddrCtlSize uintptr = 32
 
 func openDev(config Config) (ifce *Interface, err error) {
+	if config.Driver == TunTapOSXDriver {
+		return openDevTunTapOSX(config)
+	} else if config.Driver != SystemDriver {
+		return nil, errors.New("unrecognized driver")
+	}
 	if config.DeviceType != TUN {
-		return nil, errors.New("only tun is implemented on this platform")
+		return nil, errors.New("only tun is implemented for SystemDriver, use TunTapOSXDriver for tap")
```

Comment:
Tiny nit: I'd prefer having this function renamed to `openDevSystem`, then a new `openDev` that does a switch/case:

```
switch {
  case config.Driver == TunTapOSXDriver:
    return openDevTunTapOSX(config)
  case config.Driver == SyntemDriver && config.DeviceType == TUN:
    return openDevSystem(config)
  default:
    return errors.New("...")
}
```